### ISSUE TITLE
Implement VV's naming suggestion for categories 

### ISSUE THREAD

**Issue by benediktahrens** [UniMath/UniMath#699]
as per https://github.com/UniMath/UniMath/issues/362#issuecomment-225556634

**Comment by benediktahrens** [comment]
@peterlefanulumsdaine and I would like to move forward on this proposal in the next few days (hours?). I will try to merge PRs before this change, to avoid creating merge conflicts. 
Any comments and suggestions amending the original proposal are welcome.

**Comment by benediktahrens** [comment]
I'll merge 
- https://github.com/UniMath/UniMath/pull/607
- https://github.com/UniMath/UniMath/pull/632
- https://github.com/UniMath/UniMath/pull/679
before starting the renaming. The other PRs are not ready to be merged, or are not affected.

**Comment by benediktahrens** [comment]
Renaming `(is_)category` into `(is_)univalent[_category]` is done in #701 .

I'd be happy to have a short discussion about whether
`Precategory` -> `category`
is the best we can come up with.

**Comment by DanGrayson** [comment]
I think `Precategory` -> `category` or `Category` is pretty good, since it's close to what one would get by translating the mathematical definition to our context.  The only thing omitted is the condition that the objects form a set, which we can see now, from the univalent point of view, would be a mistake to perpetuate, and `setcategory` or `SetCategory` will take care of that.

(I prefer type names to be capitalized.)

**Comment by peterlefanulumsdaine** [comment]
> The only thing omitted is the condition that the objects form a set, which we can see now, from the univalent point of view, would be a mistake to perpetuate

I wouldn’t even feel this is a change/omission.  Objects forming a set (in the h-set sense, not the size sense) isn’t a condition in the standard definition, it’s just a consequence of the classical setting.  Whether it’s omitted or included, this definition becomes equivalent to the standard one under the interpretation of types as sets.

So I also feel `Precategory` -> `category` or `Category` is very good.  (Like Dan, I like capitalisation for type names.)

**Comment by DanGrayson** [comment]
#176 presents a uniform naming proposal

**Comment by DanGrayson** [comment]
I agree with Peter.  One may formulate a criterion for deciding whether a type in a translated definition should be a set: if one wants to consider equality of its elements.  In category theory, it is commonly known that considering equality of objects should be avoided, whereas considering equality of maps is unavoidable, because of associativity.

**Comment by mortberg** [comment]
Like I wrote in some other thread yesterday I agree with Dan and Peter about changing `Precategory` -> `category` or `Category`. I have no strong opinions about capitalisation for type names, but if we go for it I think we should strive for making it uniform (at least throughout the CT library). Having a predictable and consistent naming convention is important and makes the library a lot easier to use.

Another thing I noticed is that it is a bit annoying to use `Precategory` because so many of the basic results are stated with the homset property separate, this means that whenever I apply a lemma stated this way to a `Precategory` I have to extract the homset assumption explicitly. This gets quite annoying when one has to do it all the time. I would suggest we do another PR later and go over some of the basic files (`precategories.v` and `functor_categories.v`) and change every lemma parametrized by `C : precategory` and `has_homsets C` to `C : Precategory`. This change will affect a lot of code and it will be easy (but very boring) to do so I suggest doing it in a separate PR after the renaming.

**Comment by benediktahrens** [comment]
On 05/05/2017 12:54 PM, Anders Mörtberg wrote:
> Like I wrote in some other thread yesterday I agree with Dan and
> Peter about changing `Precategory` -> `category` or `Category`. I
> have no strong opinions about capitalisation for type names, but if
> we go for it I think we should strive for making it uniform (at least
> throughout the CT library). Having a predictable and consistent
> naming convention is important and makes the library a lot easier to
> use.
> 
> Another thing I noticed is that it is a bit annoying to use
> `Precategory` because so many of the basic results are stated with
> the homset property separate, this means that whenever I apply a
> lemma stated this way to a `Precategory` I have to extract the homset
> assumption explicitly. This gets quite annoying when one has to do it
> all the time. I would suggest we do another PR later and go over some
> of the basic files (`precategories.v` and `functor_categories.v`) and
> change every lemma parametrized by `C : precategory` and `has_homsets
> C` to `C : Precategory`. This change will affect a lot of code and it
> will be easy (but very boring) to do so I suggest doing it in a
> separate PR after the renaming.

I agree with everything you say!

**Comment by peterlefanulumsdaine** [comment]
I also agree with Anders’ points!

**Comment by benediktahrens** [comment]
There are a few files whose name mentions precategories in some way:

- precategories.v
- ProductPrecategory.v
- PrecategoryBinProduct.v
- PrecategoriesWithAbgrops.v
- DiscretePrecategory.v

I suggest renaming those files, replacing any variant of "precategory" by "Category".

**Comment by DanGrayson** [comment]
Sounds good.

**Comment by vladimirias** [comment]
While I said that I can live with Precategory for the time being, I do continue to think that “category” is better.

**Comment by benediktahrens** [comment]
On 05/07/2017 02:53 PM, Vladimir Voevodsky wrote:
> While I said that I can live with Precategory for the time being, I
> do continue to think that “category” is better.

That's great - this change has been implemented in
https://github.com/UniMath/UniMath/pull/702 .

**Comment by benediktahrens** [comment]
I consider this issue done, hence closing. If you find any artefacts of previous naming, please reopen or open a new issue.
